src/file-selector.ts

```diff
@@ -121,6 +121,10 @@ function flatten<T>(items: any[]): T[] {
 }
 
 function fromDataTransferItem(item: DataTransferItem) {
+    if ('getAsFileSystemHandle' in DataTransferItem.prototype) {
```

Comment:
Some unit tests would be nice.

Reply:
If you check, I've mocked some of the APIs in order to get some unit tests running.

Reply:
I looked into it, but it appears to be a rabbit hole. The current tests fail with `ReferenceError: DataTransferItem is not defined`, which is documented as https://github.com/jsdom/jsdom/issues/2913, and https://github.com/jsdom/jsdom/issues/2913#issuecomment-2119577248 suggests a [polyfill](https://atlassian.design/components/pragmatic-drag-and-drop/optional-packages/unit-testing/about), but installing it fails with unresolvable dependency issues. YOLO-updating all dependencies also doesn't work. I'm not familiar with Jest and don't have time to learn how to mock stuff with it. If someone else has idle cycles or experience, this would be appreciated. Else, feel free to close the PR if the tests are a required dependency. No hard feelings.

Reply:
I can try to have a look as well. As long as there are no regressions, we could probably merge without tests. Let me see what I can find.

I appreciate the contribution @tomayac